### PR TITLE
nixos/switch-to-configuration: fix units starting with dash not being able to reload

### PR DESCRIPTION
**Pull request by Weathercold**
Fix units starting with `-` being recognized as options, preventing them from reloading:
```console
sudo nixos-rebuild switch                                                                                                                         ~
building Nix...
building the system configuration...
activating the configuration...
setting up /etc...
reloading user units for weathercold...
setting up tmpfiles
'/nix/store/75zlhlgv7bsdgivn2i6xb5la1qdr5rpf-system-path/bin/busctl --json=short call org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager ListUnitsByNames as 1 -.mount' exited with value 1 at /nix/store/zaxzsq8aprz2qr3ph6ksd9rsjzhm1vm4-nixos-system-nixos-22.11pre399062.3a11db5f408/bin/switch-to-configuration line 140.
warning: error(s) occurred while switching to the new configuration
```
```console
/nix/store/75zlhlgv7bsdgivn2i6xb5la1qdr5rpf-system-path/bin/busctl --json=short call org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager ListUnitsByNames as 1 -.mount
/nix/store/75zlhlgv7bsdgivn2i6xb5la1qdr5rpf-system-path/bin/busctl: invalid option -- '.'
```

This is a regression introduced by #185067.

###### Description of changes
Add `--` before `$unit_name` so it is always treated as an argument.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Unfortunately I don't have time to test.

### CODE REVIEW

**Review by dasJ — COMMENTED** [review]
Change looks good, but could you add a regression test to switchTest so that this won't happen again?

**Comment by Weathercold** [comment]
Ok. I have to sleep now so I will add a test tomorrow

**Comment by Weathercold** [comment]
Done 👌

I ran the test multiple times and it failed/succeeded as expected.

**Review by stigtsp — APPROVED** [review]
LGTM, tests pass and change looks good. Thanks for fixing this!

File: nixos/tests/switch-test.nix [hunk 1]
```diff
@@ -828,6 +865,7 @@ in {
         assert_lacks(out, "restarting the following units:")
         assert_contains(out, "\nstarting the following units: no-restart-service.service, reload-triggers-and-restart-by-as.service, simple-reload-service.service, simple-restart-service.service, simple-service.service\n")
         assert_contains(out, "the following new units were started: no-restart-service.service, reload-triggers-and-restart-by-as.service, reload-triggers-and-restart.service, reload-triggers.service, simple-reload-service.service, simple-restart-service.service, simple-service.service\n")
+
```
Comment (stigtsp) [inline nixos/tests/switch-test.nix:868]: Remove added whitespace

File: nixos/tests/switch-test.nix [hunk 2]
```diff
@@ -837,6 +875,7 @@ in {
         assert_contains(out, "restarting the following units: reload-triggers-and-restart-by-as.service, simple-restart-service.service, simple-service.service\n")
         assert_lacks(out, "\nstarting the following units:")
         assert_lacks(out, "the following new units were started:")
+
```
Comment (stigtsp) [inline nixos/tests/switch-test.nix:878]: Remove added whitespace

File: nixos/tests/switch-test.nix [hunk 3]
```diff
@@ -923,6 +963,7 @@ in {
         switch_to_specialisation("${machine}", "mount")
         out = machine.succeed("mount | grep 'on /testmount'")
         assert_contains(out, "size=1024k")
+
```
Comment (stigtsp) [inline nixos/tests/switch-test.nix:966]: Remove added whitespace

File: nixos/tests/switch-test.nix [hunk 4]
```diff
@@ -846,6 +885,7 @@ in {
         assert_contains(out, "restarting the following units: reload-triggers-and-restart-by-as.service, reload-triggers-and-restart.service, simple-restart-service.service, simple-service.service\n")
         assert_lacks(out, "\nstarting the following units:")
         assert_lacks(out, "the following new units were started:")
+
```
Comment (stigtsp) [inline nixos/tests/switch-test.nix:888]: Remove added whitespace

**Review by stigtsp — COMMENTED** [review]
Just a minor nickpick: The change adds a couple of extra newlines to the test file, since they are not relevant for the change I'd prefer if they were not included in the PR.

**Comment by dasJ** [comment]
@GrahamcOfBorg test switchTest

**Comment by stigtsp** [comment]
@Weathercold Thanks again, and welcome to the NixOS project!

**Pull request merged**